PhpAmqpLib/Wire/IO/StreamIO.php

```diff
@@ -103,6 +107,48 @@ public function connect()
     {
         $errstr = $errno = null;
 
+        do {
+            if ($this->context) {
```

Comment:
I don't understand, is this code duplicating code from 

here: https://github.com/videlalvaro/php-amqplib/blob/master/PhpAmqpLib/Wire/IO/StreamIO.php#L106
here: https://github.com/videlalvaro/php-amqplib/blob/master/PhpAmqpLib/Wire/IO/StreamIO.php#L115

and ignoring the code from here? https://github.com/videlalvaro/php-amqplib/blob/master/PhpAmqpLib/Wire/IO/StreamIO.php#L85